### PR TITLE
Demo Data for new patient

### PR DESCRIPTION
**Pull request by pmadria**
A button fills the fields with values random values that are populated for the purposes of a fake/demo patient

### CODE REVIEW

**Comment by danlee1025** [comment]
image

I see a row of patients at the top, looks like it's duplicating the patients in the list.

File: app/views/patients/new.html.erb [hunk 1]
```diff
@@ -4,4 +4,6 @@
   </div>
 </section>
 
+<%= button_to 'Demo Data', new_patient_path, params: fake_patient_params, method: :get, class: 'button is-info is-light' %>
 <%= render 'form', patient: @patient, return_path: patients_path %>
+%>
```
Comment (danlee1025) [inline app/views/patients/new.html.erb:9]: I think this `%>` can be removed, doesn't seem to be matching with any `<%=`

File: app/views/patients/index.html.erb [hunk 2]
```diff
@@ -1,24 +1,24 @@
 <section class="hero is-small is-info">
-  <div class="hero-body">
-    <p class="title">Patients</p>
-  </div>
+<div class="hero-body">
+  <p class="title">Patients</p>
+</div>
```
Comment (danlee1025) [inline app/views/patients/index.html.erb:4]: lines 2-4 seem to missing an indentation.

File: app/views/patients/index.html.erb [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+        <tr>
+          <td><%= link_to patient.full_name || "((Unknown))", patient %></td>
+          <td><%= link_to 'Edit', edit_patient_path(patient), class: 'button is-info is-light is-small' %></td>
+          <td><%= link_to 'Destroy', patient, method: :delete, data: { confirm: 'Are you sure?' }, class: 'button is-danger is-outlined is-small' %></td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+  <%= link_to 'New Patient', new_patient_path, class: 'button is-info' %>
+</div>
```
Comment (danlee1025) [inline app/views/patients/index.html.erb:23]: lines 10-23 also seem to be missing an indentation.

File: app/controllers/patients_controller.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require 'patients_helper'
```
Comment (danlee1025) [inline app/controllers/patients_controller.rb:3]: I don't think this `require` is needed.

**Review by danlee1025 — COMMENTED** [review]
Demo button works as expected, just some small changes requested.

File: app/views/patients/_form.html.erb [hunk 5]
```diff
@@ -44,7 +44,6 @@
           <%= form.date_field :birth_date,
            class: 'input',
            placeholder: 'mm/dd/yyyy',
-           value: format_date(patient.birth_date),
```
Comment (Jammjammjamm) [inline app/views/patients/_form.html.erb:47]: This needs to be fixed so that it works in Safari.

File: app/views/patients/new.html.erb [hunk 6]
```diff
@@ -4,4 +4,5 @@
   </div>
 </section>
 
+<%= button_to 'Demo Data', new_patient_path, params: fake_patient_params, method: :get, class: 'button is-info is-light' %>
```
Comment (Jammjammjamm) [inline app/views/patients/new.html.erb:7]: This button needs some padding. I think the best way would be to move it into the patient form:
```
<section class="section">
  <div class="container">
    <% if current_page? new_patient_path %>
      <div class="block">
        <%= button_to 'Demo Data',
          new_patient_path,
          params: fake_patient_params,
          method: :get,
          class: 'button is-info is-light'
        %>
      </div>
    <% end %>
    <%= form_with(model: patient) do |form| %>
...
```

File: app/views/patients/_form.html.erb (hunk 5 above)
Comment (pmadria) [reply app/views/patients/_form.html.erb:47]: Implemented a change...seems to resolve in both but should I focus on a different solution?

**Review by Jammjammjamm — APPROVED** [review]

**Pull request merged**